language/move-lang/src/typing/translate.rs

```diff
@@ -73,6 +73,17 @@ fn main_function(
 fn main_function_(context: &mut Context, name: FunctionName, f: N::Function) -> T::Function {
     let loc = name.loc();
     let fdef = function(context, name, f);
+    for tparam in &fdef.signature.type_parameters {
+        match tparam.kind.value {
+            Kind_::Unknown => {
+                check_copyable_main_type_arg_error(context, loc, tparam.kind.loc, false)
+            }
+            Kind_::Resource => {
+                check_copyable_main_type_arg_error(context, loc, tparam.kind.loc, true)
+            }
+            Kind_::Unrestricted | Kind_::Affine => (),
+        }
+    }
```

Comment:
Ditto above. I don't think we need this restriction

Reply:
I think you are right. I confused myself with "scripts shall not take arguments whose types are not of kind resource".  I'll remove the restriction. Thanks for catching this!